File: backend/app/schemas/analytics.py
"""Analytics, Dashboard KPIs, and Stream Telemetry Schemas."""

from typing import Optional, Dict, Any, List
from pydantic import BaseModel


class DashboardSummaryKPIs(BaseModel):
    total_transactions_24h: int
    total_volume_usd_24h: float
    fraud_prevented_usd_24h: float
    fraud_rate_percentage: float
    active_threat_level: str  # NORMAL, ELEVATED, HIGH, SEVERE
    open_cases_count: int
    avg_inference_latency_ms: float
    p99_inference_latency_ms: float
    system_tps: float


class HourlyFraudTrend(BaseModel):
    hour: str
    total_count: int
    fraud_count: int
    volume_usd: float
    blocked_volume_usd: float


class MerchantRiskProfileDTO(BaseModel):
    merchant_id: str
    name: str
    category: str
    risk_score: float
    fraud_rate: float
    total_volume: float
    is_blacklisted: bool


class GeoRiskMetricDTO(BaseModel):
    country_code: str
    country_name: str
    risk_score: float
    transaction_count: int
    fraud_count: int


class SimulationControlRequest(BaseModel):
    action: str  # START, STOP, PAUSE, RESUME, SET_SPEED, INJECT_ATTACK
    target_tps: Optional[int] = 5
    attack_type: Optional[str] = None  # CARD_TESTING, IMPOSSIBLE_TRAVEL, ATO, CRYPTO_SURGE
    duration_seconds: Optional[int] = 30

File: frontend/src/types/analytics.ts
export interface DashboardKPIs {
  total_transactions_24h: number;
  total_volume_usd_24h: number;
  fraud_prevented_usd_24h: number;
  fraud_rate_percentage: number;
  active_threat_level: "NORMAL" | "ELEVATED" | "HIGH" | "SEVERE";
  open_cases_count: number;
  avg_inference_latency_ms: number;
  p99_inference_latency_ms: number;
  system_tps: number;
}

export interface HourlyTrendPoint {
  hour: string;
  total_count: number;
  fraud_count: number;
  volume_usd: number;
  blocked_volume_usd: number;
}

export interface MerchantRiskItem {
  merchant_id: string;
  name: string;
  category: string;
  risk_score: number;
  fraud_rate: number;
  total_volume: number;
  is_blacklisted: boolean;
}

export interface GeoRiskItem {
  country_code: string;
  country_name: string;
  risk_score: number;
  transaction_count: number;
  fraud_count: number;
}
